rigs/acme-classic-rig/profiles/library/config/rush-project.json

```diff
@@ -1,3 +1,14 @@
 {
-  "projectOutputFolderNames": ["lib", "lib-commonjs", "dist", "temp"]
+  "projectOutputFolderNames": {
+    "_phase:compile": ["lib", "lib-commonjs", "dist", "temp/compile"],
+    "_phase:lint": ["temp/lint"],
+    "_phase:test": ["temp/test"],
+    "_phase:readme_updater": ["temp/readme_updater"],
+    "_phase:push_notes": ["temp/push_notes"]
```

Comment:
Config files are more intuitive if:
- Predefined structure goes in JSON object keys  (`"projectOutputFolderNames"`,  `"associatedPhases"`, etc)
- The user's data goes in JSON values (`"temp/lint"`, `"_phase:compile"`, etc)

Because the above design doesn't follow that convention, a newcomer may wonder whether `"_phase:push_notes"` is a built-in feature of Rush, that they should search for in our docs -- or is it just example data?  It is example data.  But even if we know that, the role is still unclear:  Is `"_phase:push_notes"` the project output folder name?  No, `"temp/push_notes"` seems like a folder name, so then is `"_phase:push_notes"` the corresponding "project" name?  No.  It's a "phase name," but that terminology doesn't appear anywhere in the file format.

If we instead move the data into JSON values, we're forced to introduce a JSON key name that clears up these questions:
```js
{
  "phaseOptions": [
    {
      "phaseName": "_phase:compile",
      "outputFolderNames": ["lib", "lib-commonjs", "dist", "temp/compile"]
    },
    {
      "phaseName": "_phase:lint",
      "outputFolderNames": ["temp/lint"]
    },
    {
      "phaseName": "_phase:test",
      "outputFolderNames": ["temp/test"]
    },
    {
      "phaseName": "_phase:readme_updater",
      "outputFolderNames": ["temp/readme_updater"]
    },
    {
      "phaseName": "_phase:push_notes",
      "outputFolderNames": ["temp/push_notes"]
    }
  ]
}
```


People will be quick to point out some downsides of this approach:
- The person writing the config file has to write more characters
- If the same `"phaseName"` appears twice, that won't be caught for free as a syntax error (`JSON object has a duplicate key "_phase:push_notes"`).  Instead we have to implement custom validation with an arguably more intelligible message (`The phase name "_phase:push_notes" is defined twice in "phaseOptions"`)

But these are concerns of someone who is implementing the system.  The higher priority is the ***end user***, the newcomer who is trying to figure out how our stuff works, who is making a decision whether to adopt it.

`</personal-soapbox>`
`</grain-of-salt>`

Reply:
That seems reasonable -- will noodle on your suggestion.